numpy/core/arrayprint.py

```diff
@@ -1220,15 +1220,21 @@ def format_float_positional(x, precision=None, unique=True,
 
 
 class IntegerFormat:
-    def __init__(self, data):
+    def __init__(self, data, sign=False):
+        if isinstance(sign, bool):
+            sign = '+' if sign else '-'
         if data.size > 0:
             max_str_len = max(len(str(np.max(data))),
                               len(str(np.min(data))))
         else:
             max_str_len = 0
         self.format = '%{}d'.format(max_str_len)
+        self.sign = sign
 
     def __call__(self, x):
+        if self.sign == '+':
+            return '+' + str(x) if x >= 0 else self.format % x
```

Comment:
To convert `x` to a string, we do need to do the `self.format % x` in both cases, rather than using `str()`.

Reply:
Added in a conditional in `self.format`, if it receives '+' or '  ' as the `sign` it will format it as `'%{}s'.format(max_str_len)` instead.